docs/src/proposals/fee_transaction_priority.md

```diff
@@ -47,111 +47,128 @@ locking complexity.
 
-1. `default_transaction_queue` - A max-heap `BinaryHeap<Transaction>` that tracks all pending transactions.
-The priority in the heap is the additional fee of the transaction. Transactions are added to this queue
-from sigverify before the leader slot begins.
-2. `all_transaction_queues` - A `VecDeque<BinaryHeap<Transaction>>` that tracks all pending queues of work.
-Some pending queues have higher priority than others (as will be explained later in the `Handling Completion Signals from BankingStage Threads` section below). The list is ordered in priority from highest to lowest priority. On
-initialization, `all_transaction_queues[0] = default_transaction_queue`.
-3. `locked_accounts` - A `HashMap<LockedPubkey, usize>` that tracks the set of accounts needed to execute the
-current set of transactions scheduled/sent to banking threads. Accounts are added to this set
-*before* being sent to BankingStage threads. The `usize` represents a refcount, because multiple read
-accounts could be grabbed.`LockedPubkey` is defined as:
+1. `pending_transactions` - A max-heap `BinaryHeap<Transaction>` that tracks all pending transactions that are not known to be blocked.
+2. `transactions_by_account` - A map `HashMap<Pubkey, AccountTransactionQueue>` that tracks all pending and blocked transactions by account key.
+   The `AccountTransactionQueue` is defined as:
+   ```
+   class AccountTransactionQueue {
+       /// Tree of read transactions on the account ordered by fee-priority
+       pub reads: RBTree<Transaction>,
+       /// Tree of write transactions on the account ordered by fee-priority
+       pub writes: RBTree<Trasaction>,
+       /// The type, count, and minimum priority transaction that is currently scheduled for execution
+       pub scheduled_lock: Option<AccountLock>,
+   }
+   ```
+   and `AccountLock` is defined as:
+   ```
+       enum AccountLock {
+           Read(usize, Transaction),
+           Write(Transaction),
+       }
+   ```
+   where the `Transaction` stored in the `AccountLock` will be the minimum priority transaction currently scheduled. The `Read` variant will also store the number
+   of read transactions.
+3. `blocked_transactions` - A map `HashMap<Signature, Vec<Transaction>>` that maps from the blocking transaction's `Signature` to the transactions it is blocking.
+This is lazily evaluated only at the time a transaction is scheduled, or a transaction completes execution.
+
+#### Incoming Transactions:
+
+When the Scheduler receives transactions from SigVerify, the Scheduler's components need to be updated to reflect the new priority.
+For each transaction `transaction` the Scheduler receives we run the following:
 ```
-enum LockedPubkey {
-    Read(Pubkey),
-    Write(Pubkey),
-}
+    for account in transaction.accounts {
+        let mut account_queue = self.transactions_by_account.entry(account.key()).or_default();
+        match account {
+            Read(pubkey) => account_queue.reads.insert(transaction),
```

Comment:
will this be Rc<Tx>?

Reply:
Yeah probably using Rc<Transaction> (maybe Arc<Transaction> since we'll be sending to the banking stage threads?), but yeah some form of reference count